agnostics/kare-dns/k3s/base/coredns-patcher-job.yaml

```diff
@@ -0,0 +1,230 @@
+      exit 1
+    fi
+
+    echo "Waiting for kare-dns service to be ready..."
+    TIMEOUT=300
+    ELAPSED=0
+
+    while [ $ELAPSED -lt $TIMEOUT ]; do
+      KARE_DNS_IP=$(kubectl get service kare-dns-coredns -n kare-dns -o jsonpath='{.spec.clusterIP}' 2>/dev/null || echo "")
+
+      if [ -n "$KARE_DNS_IP" ] && [ "$KARE_DNS_IP" != "null" ] && [ "$KARE_DNS_IP" != "<no value>" ]; then
+        echo "Found kare-dns service ClusterIP: $KARE_DNS_IP"
+        if echo "$KARE_DNS_IP" | grep -E '^[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+$' > /dev/null; then
+          # Create Corefile content
+          {
+            echo "${ENVIRONMENT}.${DOMAIN}:53 {"
+            echo "    errors"
+            echo "    forward . $KARE_DNS_IP:53"
+            echo "}"
+            echo ".:53 {"
+            echo "    errors"
+            echo "    health"
+            echo "    ready"
+            echo "    kubernetes cluster.local in-addr.arpa ip6.arpa {"
+            echo "        pods insecure"
+            echo "        fallthrough in-addr.arpa ip6.arpa"
+            echo "    }"
+            echo "    hosts /etc/coredns/NodeHosts {"
+            echo "        ttl 60"
+            echo "        reload 15s"
+            echo "        fallthrough"
+            echo "    }"
+            echo "    prometheus :9153"
+            echo "    forward . $UPSTREAM_DNS"
+            echo "    cache 30"
+            echo "    loop"
+            echo "    reload"
+            echo "    loadbalance"
+            echo "    import /etc/coredns/custom/*.override"
+            echo "}"
+            echo "import /etc/coredns/custom/*.server"
+          } > /tmp/new-corefile
+          echo "Patching CoreDNS ConfigMap with kare-dns IP: $KARE_DNS_IP"
+
+          if kubectl patch configmap coredns -n kube-system --patch '{"data":{"Corefile":"'"$(cat /tmp/new-corefile | sed 's/$/\\n/' | tr -d '\n' | sed 's/\\n$//')"'"}}'; then
+            echo "Successfully patched CoreDNS ConfigMap"
+
+            echo "Restarting CoreDNS deployment..."
+            if kubectl rollout restart deployment/coredns -n kube-system; then
+              echo "Successfully restarted CoreDNS deployment"
+
+              echo "Waiting for CoreDNS rollout to complete..."
+              if kubectl rollout status deployment/coredns -n kube-system --timeout=120s; then
+                echo "CoreDNS configuration updated successfully!"
+                exit 0
+              else
+                echo "WARNING: CoreDNS rollout timed out, but patch was applied"
+                exit 0
+              fi
+            else
+              echo "ERROR: Failed to restart CoreDNS deployment"
+              exit 1
+            fi
+          else
+            echo "ERROR: Failed to patch CoreDNS ConfigMap"
+            exit 1
+          fi
+        else
+          echo "Invalid IP format: $KARE_DNS_IP, retrying..."
+        fi
+      fi
+
+      echo "kare-dns service not ready yet, waiting..."
+      sleep 5
```

Comment:
Very minor point, but if you put the sleep at the start of the loop instead of the end you'll use the full timeout, at the moment on the last check you'll wait 5 seconds, then fail instead of rechecking before failing.

Same for coredns-kare-dns-patcher below.